lib/rules/use-isnan.js

```diff
@@ -82,7 +86,30 @@ module.exports = {
                 /^(?:[<>]|[!=]=)=?$/u.test(node.operator) &&
                 (isNaNIdentifier(node.left) || isNaNIdentifier(node.right))
             ) {
-                context.report({ node, messageId: "comparisonWithNaN" });
+                const isFixable = fixableOperators.has(node.operator);
+
+                if (isFixable) {
+                    context.report({
+                        node,
+                        messageId: "comparisonWithNaN",
+                        suggest: [
+                            {
+                                messageId: "replaceWithIsNaN",
+                                fix(fixer) {
+                                    const comparedValue = isNaNIdentifier(node.left) ? node.right : node.left;
+                                    const shouldNegate = node.operator[0] === "!";
+
+                                    const negation = shouldNegate ? "!" : "";
+                                    const comparedValueText = sourceCode.getText(comparedValue);
+
+                                    return fixer.replaceText(node, `${negation}Number.isNaN(${comparedValueText})`);
```

Comment:
If `comparedValue` is a SequenceExpression, we should wrap it in parentheses.

For example:

```js
(foo, bar) === NaN // looks like intent to check if `bar` is NaN
```

Current suggestion:

```js
Number.isNaN(foo, bar) // incorrect, this evaluates to Number.isNaN(foo)  
```

It should be:

```js
Number.isNaN((foo, bar)) // correct, this evaluates to Number.isNaN(bar)  
```

An example of how this could be implemented:

https://github.com/eslint/eslint/blob/1fedfd28a46d86b2fbcf06a2328befafd6535a88/lib/rules/no-prototype-builtins.js#L132-L134

Reply:
Oh, that's a good one. Forgot it exists (I literally had the same issue in ts-eslint 2 weeks ago 😂)
Fixed